pkg/controller/instance/controller_reconcile.go

```diff
@@ -142,9 +142,9 @@ func (igr *instanceGraphReconciler) reconcileResource(ctx context.Context, resou
 	resourceState := &ResourceState{State: "IN_PROGRESS"}
 	igr.state.ResourceStates[resourceID] = resourceState
 
-	// Check if resource should be created
-	if want, err := igr.runtime.WantToCreateResource(resourceID); err != nil || !want {
-		log.V(1).Info("Skipping resource creation", "reason", err)
+	// Check if resource should be processed (create or get)
+	if want, err := igr.runtime.ReadyToProcessResource(resourceID); err != nil || !want {
```

Comment:
I do sorta wonder if we want something _other_ than template (but which we also template?) `externalRef` that only has `name`/`namespace`/`kind`/`apiversion`.

I am also fine keeping it `template` as well, I don't feel super strongly about the above.

Reply:
I can add a validation that allows only name/namespace/kind/apiVersion in the template field. That way we keep the definition tight and set the correct expectation as well. WDYT ?

Reply:
will make it a typed struct parallel to template